core/checker.go

```diff
@@ -187,6 +196,78 @@ func (s *Service) checkTcp(record bool) *Service {
 	return s
 }
 
+// checkGrpc will check a GRPC service
+func (s *Service) checkGrpc(record bool) *Service {
+	dnsLookup, err := s.dnsCheck()
+	if err != nil {
+		if record {
+			recordFailure(s, fmt.Sprintf("Could not get IP address for domain %v, %v", s.Domain, err))
+		}
+		return s
+	}
+	s.PingTime = dnsLookup
+	t1 := time.Now()
+	timeout := time.Duration(s.Timeout) * time.Second
+
+	grpcOption := grpc.WithInsecure()
+
+	// Check if TLS
+	if s.VerifySSL.Bool {
+		h2creds := credentials.NewTLS(&tls.Config{NextProtos: []string{"h2"}})
+		grpcOption = grpc.WithTransportCredentials(h2creds)
+	}
+
+	// Set up a connection to the server.
+	conn, err := grpc.Dial(s.Domain+":"+strconv.Itoa(s.Port), grpcOption)
+	if err != nil {
+		if record {
+			recordFailure(s, fmt.Sprintf("GRPC Error %v", err))
+		}
+		return s
+	}
+	defer conn.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	c := healthpb.NewHealthClient(conn)
+	in := &healthpb.HealthCheckRequest{}
+	res, err := c.Check(ctx, in)
+	if err != nil {
+		if record {
+			recordFailure(s, fmt.Sprintf("GRPC Error %v", err))
+		}
+		return s
+	}
+	t2 := time.Now()
+	s.Latency = t2.Sub(t1).Seconds()
+	s.LastResponse = res.String()
+	s.LastStatusCode = int(res.GetStatus())
+	s.ExpectedStatus = 1
+	s.Expected.String = "SERVING"
+
+	match, err := regexp.MatchString(s.Expected.String, res.String())
+	if err != nil {
+		log.Warnln(fmt.Sprintf("Service %v expected: %v to match %v", s.Name, res.String(), s.Expected.String))
+	}
+	if !match {
+		if record {
+			recordFailure(s, fmt.Sprintf("GRPC Response Body did not match '%v'", s.Expected))
+		}
+		return s
+	}
+	if s.ExpectedStatus != int(res.Status) {
+		if record {
+			recordFailure(s, fmt.Sprintf("GRPC Status Code %v did not match %v", res.Status, healthpb.HealthCheckResponse_ServingStatus(s.ExpectedStatus)))
```

Comment:
It would be nice to see something like you've done in line 255 to increase readability
https://github.com/hunterlong/statping/blob/6fd888bbaa2643a06eca32edc73fecc55f2380c1/core/checker.go#L255

Reply:
Done